src/Microsoft.ML.EntryPoints/PermutationFeatureImportance.cs

```diff
@@ -0,0 +1,411 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML;
+using Microsoft.ML.CommandLine;
+using Microsoft.ML.Data;
+using Microsoft.ML.EntryPoints;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Transforms;
+
+[assembly: LoadableClass(typeof(void), typeof(PermutationFeatureImportanceEntryPoints), null, typeof(SignatureEntryPointModule), "PermutationFeatureImportance")]
+
+namespace Microsoft.ML.Transforms
+{
+    internal static class PermutationFeatureImportanceEntryPoints
+    {
+        [TlcModule.EntryPoint(Name = "Transforms.PermutationFeatureImportance", Desc = "Permutation Feature Importance (PFI)", UserName = "PFI", ShortName = "PFI")]
+        public static PermutationFeatureImportanceOutput PermutationFeatureImportance(IHostEnvironment env, PermutationFeatureImportanceArguments input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            var host = env.Register("Pfi");
+            host.CheckValue(input, nameof(input));
+            EntryPointUtils.CheckInputArgs(host, input);
+
+            var mlContext = new MLContext();
```

Comment:
You shouldn't be creating a new MLContext here. You should use the environment being passed into the method.

Reply:
`IHostEnvironment` doesn't have model loading that I need here. Casting it to `MLContext` in the method body returns null due to inheritance direction. Also, changing the type in the method signature from `IHostEnvironment` to `MLContext` doesn't work as the first parameter is required to be `IHostEnvironment` here:
https://github.com/dotnet/machinelearning/blob/edfd10f131abe97a7f7bb25c0070843a67b8b660/src/Microsoft.ML.Core/ComponentModel/ComponentCatalog.cs#L310-L311

What do you suggest here? Change the component catalog code above to accept both `IHostEnvironment` and `MLContext`, and change the type in the method signature ?

Reply:
> What do you suggest here? 

I think one of the following approaches would work just fine:

1. Refactor `ModelOperationsCatalog.Load` to be an internal static method that takes in the `IHostEnvironment` along with the `Stream`. Call it here and in the original `ModelOperationsCatalog.Load` method.
2. Create a `new ModelOperationsCatalog` instance with the `IHostEnvironment` you have, and call `Load` using it. No need to create a new `MLContext`.

We can't be creating new environments in the middle of a library function. We need to use the environment coming from the caller.